Sources/StreamChatUI/CommonViews/PillButton.swift

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class PillButton: _Button, AppearanceProvider {
+    override open var isHighlighted: Bool {
+        didSet {
+            backgroundColor = isHighlighted
+                ? appearance.colorPalette.highlightedBackground
+                : appearance.colorPalette.staticColorText
+        }
+    }
+
+    // MARK: - Lifecycle
+
+    override open func setUpAppearance() {
+        super.setUpAppearance()
+
+        tintColor = appearance.colorPalette.staticBlackColorText
+        setTitleColor(appearance.colorPalette.staticBlackColorText, for: .normal)
+        backgroundColor = isHighlighted
+            ? appearance.colorPalette.highlightedBackground
+            : appearance.colorPalette.staticColorText
```

Comment:
Shouldn't we create a new entry for this? We are using a property that is supposed to be related to text

Reply:
I believe that the mistake is in the name of variable and not the usage. ColorPalettes shouldn't be semantic (Themes can be semantic) to allow flexibility. That being said I'm seeing that we are using `text` colours as non-text colours in various cases (e.g. ChatChannelListErrorView.backgroundColor, ChatMessageDeliveryStatusCheckmarkView.tintColor etc), so we are not really off here. Let's see what @nuno-vieira will also say and we can decide how to move forward. WDYT?

Reply:
The `ColorPallete` is a bit all over the place right now. And the name is probably misleading, as it is probably more a "Theme" than a "ColorPallete". The problem with the current structure is that customers find it hard to customize because they don't have enough context on the names of the colors. So we did discuss recently that we want to provide more context on the names of the colours. [It is also something we want to add to change on v5](https://www.notion.so/stream-wiki/StreamChat-v5-e7cd0c287fba41e8af69ae507c40d9c9?pvs=4#c70a12dfe5484a359b7aa4895a4a8183). But @ipavlidakis is correct, that for a ColorPallete, the name of this color is incorrect. I don't mind leaving it like this for now, since other views are doing the same, and the customer can still customize this in the view. But indeed what we want in the future is to start adding more context to the colors and move away from the "ColorPallete" approach.